### PR TITLE
Xcode 10 beta 1 compatibility

### PR DESCRIPTION
**Pull request by chrisballinger**
This fixes warnings and compiler errors generated by Xcode 10 when running in Swift 4.1 mode only. The Swift 4.2 migration will be a lot heavier because a lot of SDK stuff has been renamed, which will require a lot of `#if swift(>= 4.2)` to retain backwards compatibility.

Note: Adding `@escaping FrameChangeHandler` is a minor breaking change for consumers of the library.

### CODE REVIEW

File: Pod/Classes/Acknowledgements/AcknowlegementsListViewModel.swift [hunk 1]
```diff
@@ -73,7 +73,11 @@ private extension AcknowledgementsListViewModel {
         // First and last elements in the settings bundle are not needed (title and empty row).
         // these are broken up to speed up compile times
         let innerRange = 1..<(specifiers.count - 1)
+        #if swift(>=4.1)
+        let rawAcknowledgements = specifiers[innerRange].compactMap(AcknowledgementViewModel.init(dictionary:))
```
Comment (ZevEisenberg) [inline Pod/Classes/Acknowledgements/AcknowlegementsListViewModel.swift]: #pp define a compatibility alias for `compactMap` on Swift 3.x, like we did in BonMot, so this line doesn't have to be conditionally compiled

File: Pod/Classes/Keyboard/Keyboard.swift [hunk 2]
```diff
@@ -28,7 +28,7 @@ public class Keyboard {
      - parameter animated: Whether or not to animate changes in the handler block alongside the keyboard frame changes.
      - parameter handler:  A block in which to perform view changes.
      */
-    public static func addFrameObserver(_ observer: AnyObject, withAnimations animated: Bool = true, handler: FrameChangeHandler) {
+    public static func addFrameObserver(_ observer: AnyObject, withAnimations animated: Bool = true, handler: @escaping FrameChangeHandler) {
```
Comment (ZevEisenberg) [inline Pod/Classes/Keyboard/Keyboard.swift]: #? Why the change here? Does Swift 4.1 reject this, but earlier versions didn't? Regardless, no worries about semver, since we're either not using it for Swiftilities, or we're below 1.0 so it doesn't matter. Either way, breaking changes are fine.

**Comment by chrisballinger** [comment]
@ZevEisenberg  Sigh. So I couldn't resolve why the tests were failing from `scripts/iOS.sh` but not within Xcode itself. I switched over to running the tests via fastlane instead and everything seems to be working now.

```
2018-06-20 12:26:48.650 xcodebuild[1047:15407] Error Domain=IDETestOperationsObserverErrorDomain Code=6 "Early unexpected exit, operation never finished bootstrapping - no restart will be attempted" UserInfo={NSLocalizedDescription=Early unexpected exit, operation never finished bootstrapping - no restart will be attempted}
2018-06-20 12:26:48.651 xcodebuild[1047:17399] Connection peer refused channel request for "dtxproxy:XCTestManager_IDEInterface:XCTestManager_DaemonConnectionInterface"; channel canceled <DTXChannel: 0x7fc7eb728160>

Testing failed:
    Early unexpected exit, operation never finished bootstrapping - no restart will be attempted
** TEST FAILED **
```

This PR now also switches the default CircleCI environment to Xcode 9.4, and runs some additional checks with 9.2.0 to verify that the compatibility shim is working properly.
